### PR TITLE
[AC-1741] Include owners/admins can manage all collections setting in license file

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Collection Management Settings should be added to the license file and should be updated in the self-hosted instance when the license file is updated.

This PR is dependent on https://github.com/bitwarden/server/pull/3379

## Code changes

* **src/Api/AdminConsole/Controllers/OrganizationsController.cs:** Updated endpoint to allow only cloud access.
* **src/Core/Entities/Organization.cs:** Added `AllowAdminAccessToAllCollectionItems` from license.
* **src/Core/Models/Business/OrganizationLicense.cs:** Added `AllowAdminAccessToAllCollectionItems` property to OrganizationLicense and incremented version numbering
* **src/Core/Services/Implementations/OrganizationService.cs** Sets property if the feature flag is enabled or defaults to true

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by eliykat** [comment]
Billing Team should own license files so I've requested a review from them, please treat them as codeowners as well.

File: src/Core/Models/Business/OrganizationLicense.cs
Comment (cyprain-okeke) [inline src/Core/Models/Business/OrganizationLicense.cs:148]: Is this comment still validate or we need to update it?

**Review by cyprain-okeke — COMMENTED** [review]
Looks Good

File: src/Api/AdminConsole/Controllers/OrganizationsController.cs
Comment (cyprain-okeke) [inline src/Api/AdminConsole/Controllers/OrganizationsController.cs:784]: should the user be able to change this setting on the self-hosted instance?

Comment (gbubemismith) [reply src/Api/AdminConsole/Controllers/OrganizationsController.cs:784]: Only cloud access is required of this

File: src/Core/Models/Business/OrganizationLicense.cs
Comment (gbubemismith) [reply src/Core/Models/Business/OrganizationLicense.cs:148]: This is still valid

**Comment by eliykat** [comment]
This needs to wait until #3388 is merged, which adds the first collection management setting (limit collection creation/deletion) to the license file. Currently, you're both trying to bump the license version at the same time. And both PRs should wait until unit tests in #3474 are merged.

File: util/Migrator/DbScripts/2023-10-24_04_AdminCollectionItemAccess.sql
Comment (eliykat) [inline util/Migrator/DbScripts/2023-10-24_04_AdminCollectionItemAccess.sql:1]: Looks like this was accidentally included

File: src/Core/Models/Business/OrganizationLicense.cs
Comment (eliykat) [inline src/Core/Models/Business/OrganizationLicense.cs:358]: Should be commented our per the version 13 validation above.

This is because self-hosted orgs do not use this value unless the feature flag is turned on in the self-hosted instance, therefore we expect this check to fail in some circumstances. (e.g. cloud has the flag on, client updates the setting and then updates their license file, but the flag is off in self-hosted. The license value will be ignored, the org value won't match the license file, so this validation would otherwise fail.)

We can do this because validation of this property is not particularly important either (it's not billable).

File: src/Core/Models/Business/OrganizationLicense.cs
Comment (eliykat) [inline src/Core/Models/Business/OrganizationLicense.cs:141]: Needs to default to `true`. This covers the case where a client loads an old license file which doesn't have this value.

**Review by eliykat — CHANGES_REQUESTED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

File: src/Core/AdminConsole/Entities/Organization.cs
Comment (eliykat) [inline src/Core/AdminConsole/Entities/Organization.cs:271]: The naming here isn't very clear, sorry. `flexibleCollectionsIsEnabled` refers to the MVP feature flag. But this is a v1 feature. So you'll need to pass in both:

```cs
public void UpdateFromLicense(OrganizationLicense license, bool flexibleCollectionsMvpIsEnabled, bool flexibleCollectionsV1IsEnabled)
{
   // snip
       LimitCollectionCreationDeletion = !flexibleCollectionsMvpIsEnabled || license.LimitCollectionCreationDeletion;
       AllowAdminAccessToAllCollectionItems = !flexibleCollectionsV1IsEnabled || license.AllowAdminAccessToAllCollectionItems;
```

It's awkward but we'll clean it up after release.

File: src/Core/AdminConsole/Services/Implementations/OrganizationService.cs
Comment (eliykat) [inline src/Core/AdminConsole/Services/Implementations/OrganizationService.cs:605]: Same here, `AllowAdminAccessToAllCollectionItems` should use the v1 feature flag, and please rename existing variable to clearly distinguish between them.

`LimitCollectionCreationDeletion` should continue to use the mvp feature flag.

**Review by eliykat — CHANGES_REQUESTED** [review]
The license file part of it looks good, thanks for taking the time to figure out those tests.

Just noticed this, then we're good to go.

**Review by eliykat — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

**Pull request merged**
